cranelift/codegen/src/isa/zkasm/lower.isle

```diff
@@ -31,15 +31,10 @@
 (rule 0 (lower (has_type (ty_int_ref_scalar_64 ty) (iadd x y)))
   (rv_add x y))
 
-;; TODO: restore it, disabled cause matches 1 as i64 but don't correspond our i32 
-;; representation.
+(rule 1 (lower (has_type (fits_in_32 (ty_int ty)) (iadd x y)))
+  (gen_andi (rv_add x y) 0xFFFFFFFF))
```

Comment:
Should we rename rv to zk? Also, very happy that now we actually benefit from using isle

Reply:
Yes, we should gradually migrate all of these ISLE rules to "zk_*" convention. I'll do this for `rv_add` in a separate PR, here is the issue to track this https://github.com/near/wasmtime/issues/156

Reply:
And yes, ISLE makes this so much easier to write - especially the register allocation part that gets auto-magically figured out.